src/oauth2c.erl

```diff
@@ -208,12 +166,13 @@ request(Method, Type, Url, Expect, Headers, Body, Client) ->
 request(Method, Type, Url, Expect, Headers, Body, Options, Client) ->
   case do_request(Method,Type,Url,Expect,Headers,Body,Options,Client) of
     {{_, 401, _, _}, Client2} ->
-      {ok, _RetrHeaders, Client3} =
-        do_retrieve_access_token(Client2, Options),
+       {ok, _RetrHeaders, Client3} = get_access_token(Client2, Options),
```

Comment:
So here we may just have user a cached token and we got back 401. It's very likely that using the same token again will render a 401. Should we ensure the cache entry is deleted before doing this second attempt?

Reply:
You are very right, I forgot that line in the last commit. I will fix it!